tasks/send-slack-notification/send-slack-notification.yaml

```diff
@@ -0,0 +1,78 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  labels:
+    app.kubernetes.io/version: "0.0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "release"
+  name: send-slack-notification
+spec:
+  description: >-
+    Sends message to slack using postMessage API
+  params:
+    - name: message
+      description: Message to be sent
+    - name: tasksStatus
+      description: status of tasks execution
+    - name: secretName
+      description: |
+        Name of secret which contains authentication token for app
+      default: slack-notification-secret
+  volumes:
+    - name: slack-token
+      secret:
+        secretName: $(params.secretName)
+  steps:
+    - name: send-message
+      image: >-
+        quay.io/hacbs-release/release-utils@sha256:5733ece907aa70d6ebced36484c936731e8b27bfcf87fed226a0ecf95047a6b8
+      volumeMounts:
+        - name: slack-token
+          mountPath: "/etc/secrets"
+          readOnly: true
+      env:
+        - name: HOME
+          value: /tekton/home
+        - name: MESSAGE
+          value: $(params.message)
+        - name: TASKSSTATUS
+          value: $(params.tasksStatus)
+      script: |
+        #!/usr/bin/env sh
+        set -eux
+
+        if [ -z "${MESSAGE/$'\n'}" ]; then
+          echo "Message is empty - skipping send to slack"
+          exit
+        fi
+
+        if [ -f "/etc/secrets/token" ]; then
+          TOKEN=$(cat etc/secrets/token)
+        else
+          echo "Secret not defined properly"
+          exit 1
+        fi
+
+        # replace @@CIRCLE_TYPE@@
+        # replace @@STATUS_TEXT@@
+        # set a default for circleType
+        circleType="white_circle"
+        if [ "${TASKSSTATUS}" == "Failed" ]; then
+          circleType="red_circle"
+        fi
+        if [ "${TASKSSTATUS}" == "Succeeded" ]; then
+          circleType="large_green_circle"
+        fi
+        MESSAGE=$(echo "${MESSAGE}" | sed "s/@@CIRCLE_TYPE@@/${circleType}/g")
+        MESSAGE=$(echo "${MESSAGE}" | sed "s/@@STATUS_TEXT@@/${TASKSSTATUS}/g")
+
+        cat > $(workspaces.data.path)/release.json << EOF
+        ${MESSAGE}
+        EOF
+
+        curl  -H "Content-type: application/json" --data-binary "@$(workspaces.data.path)/release.json" -H "Authorization: Bearer ${TOKEN}" -X POST https://slack.com/api/chat.postMessage
+  workspaces:
```

Comment:
not sure why we need this

Reply:
removed workspace

Reply:
The workspace is still showing in the latest code